orchagent/vnetorch.h

```diff
@@ -404,12 +419,46 @@ struct VNetTunnelRouteEntry
     NextHopGroupKey secondary;
 };
 
+struct VNetLocEpAclRule
+{
+    swss::IpPrefix vip;
+    swss::IpAddress nh_ip;
+    std::string rule_name;
+};
+
 typedef std::map<NextHopGroupKey, NextHopGroupInfo> VNetNextHopGroupInfoTable;
 typedef std::map<IpPrefix, VNetTunnelRouteEntry> VNetTunnelRouteTable;
 typedef std::map<IpAddress, BfdSessionInfo> BfdSessionTable;
 typedef std::map<IpPrefix, std::map<IpAddress, MonitorSessionInfo>> MonitorSessionTable;
 typedef std::map<IpAddress, VNetNextHopInfo> VNetEndpointInfoTable;
 
+class VNetTunnelTermAcl
+{
+public:
+    VNetTunnelTermAcl(DBConnector *cfgDb, DBConnector *appDb);
+
+    bool createAclRule(const string vnet_name, swss::IpPrefix& vip, swss::IpAddress nh_ip);
+    bool removeAclRule(const string vnet_name, swss::IpPrefix& vip);
+    std::function<std::string(const std::string&, const std::string&)> concat =
+        [](const std::string &a, const std::string &b) { return a + "," + b; };
+    VNetLocEpAclRule getAclRule(const string vnet_name, const swss::IpPrefix& vip);
+
+protected:
+
+    void lazyInit();
+    std::vector<std::string> getBindPoints();
+    std::set<std::string> findInternalPorts();
+    std::string getNbrAlias(const swss::IpAddress& ip);
```

Comment:
Can EniFwdCtx from [dash/dashenifwdorch.h](https://github.com/sonic-net/sonic-swss/blob/master/orchagent/dash/dashenifwdorch.h#L370) be re-used here for ease of maintenance. 

Calls like findInternalPorts, getBindPoints, getNbrAlias can all be re-used.

Reply:
Hi Vivek, I did think about reusing `EniFwdCtx` but discussed with Riff, local_endpoint support is supposed to be generic, it's a bit unnatural to include from dashenifworch for getting neighbor alias...

Reply:
In that case, i'd recommend having a new class. My reasoning being orchagent is already huge and we should try to avoid duplicate code as much as possible

```
class TunnelTermHelper 
{
public:
      virtual void initiailze();
      std::vector<std::string> getBindPoints();
      std::set<std::string> findInternalPorts();
      std::string getNbrAlias(const swss::IpAddress& ip);
      virtual std::string getRuleName(const std::string& vnet_name, const swss::IpPrefix& vip);
}
```

We can then make EniFwdCtx derive from TunnelTermHelper, making this generic. @r12f What do you think?